integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -268,6 +309,151 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
+
+        val handlerClassName: String = handlerClass match {
+          case Some(className) =>
+            className
+          case None =>
+            /* use handler type to find the default implementation */
+            if (handlerType.isEmpty) {
+              throw new MalformedCarbonCommandException(
+                s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                  s"Both ${CarbonCommonConstants.INDEX_HANDLER}.$e.class and " +
+                  s"${CarbonCommonConstants.INDEX_HANDLER}.$e.type properties are not specified")
+            } else if (handlerType.get.equalsIgnoreCase("geohash")) {
+              /* Use geoHash default implementation */
+              val className = classOf[org.apache.carbondata.core.util.GeoHashDefault].getName
+              tableProperties.put(s"${CarbonCommonConstants.INDEX_HANDLER}.$e.class", className)
+              className
+            } else {
+              throw new MalformedCarbonCommandException(
+                s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                  s"Unsupported value : ${handlerType.get} specified in property " +
+                  s"${CarbonCommonConstants.INDEX_HANDLER}.$e.type")
+            }
+        }
+
+        try {
+          val handlerClass: Class[_] = java.lang.Class.forName(handlerClassName)
+          val instance = handlerClass.newInstance().asInstanceOf[
+            CustomIndex[Long, String, java.util.List[Array[Long]]]]
+          instance.validateOption(tableProperties.asJava)
+          instance.init(tableProperties.asJava)
+        } catch {
+          case ex: ClassNotFoundException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"$handlerClassName class in property " +
+                s"${CarbonCommonConstants.INDEX_HANDLER}.$e.class is failed with $ex")
+          case ex@(_: InstantiationError | _: IllegalAccessException) =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Instantiation of class $handlerClassName is failed with $ex")
+          case ex: ClassCastException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Failed due to $ex")
+        }
+
+        /* Add target column in sort column */
+        var sortKey = tableProperties.getOrElse(CarbonCommonConstants.SORT_COLUMNS, "")
+        if (!sortKey.isEmpty) {
+          /* Source columns are not allowed to be specified in sort columns. Instead target column
+          is implicitly treated as sort column */
+          sources.foreach { column =>
+            sortKey.split(",").foreach { key =>
```

Comment:
instead of looping, you can first split and trim and then directly do contains on it.

Reply:
This code is removed now. Source columns may also be present in sortcolumns.